### DISCUSSION TITLE
textDocument/didOpen

### DISCUSSION THREAD

**Discussion by ballcoach12** [eclipse-langium/langium#1438] [Q&A] [answered]
I recall reading somewhere that Langium doesn't handle `textDocument/didOpen`, only `didChange`.  If this is correct, what is the preferred way of giving an editor a new document?  In my case, I am building a browser-based editor that doesn't run in VSCode, so I have to plumb in all the operations myself with a custom file system component that we wrote.

**Comment by msujew** [comment, marked as answer]
Hey @ballcoach12,

Langium handles both `didOpen` and `didChange` using the same callback, see [here](https://github.com/eclipse-langium/langium/blob/d873ebeb8faaf4d1ad5456461b15154dc2dc279b/packages/langium/src/lsp/language-server.ts#L270) (even though it says `onDidChangeContent` it actually fires on both events). Calling the `textDocument/didOpen` notification with the appropriate arguments should bring the desired effect.

**Reply by ballcoach12** [reply]
Pefect! Thank you, Mark!
